huggingface_pipelines/pipeline_config.py

```diff
@@ -0,0 +1,78 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from typing import List
+
+
+@dataclass
+class PipelineConfig(ABC):
+    """
+    Abstract base class for pipeline configurations.
+
+    Attributes:
+        dataset_name (str): The name of the HuggingFace dataset to be used.
+        dataset_split (str): The dataset split to be used (e.g., 'train', 'test', 'validation').
+        batch_size (int): The batch size to be used for processing.
+        num_shards (int): The number of shards to split the dataset into. Defaults to 1.
+        shard_id (int): The ID of the shard to process. Defaults to 0.
+        device (str): The device to use for inference (e.g., 'cpu', 'cuda'). Defaults to 'cpu'.
+        cache_to_arrow (bool): Whether to cache results to Arrow format. Defaults to False.
+        output_file_name (str): The base name of the file where results will be saved. Defaults to "results".
+    """
+    dataset_name: str
+    dataset_split: str
+    batch_size: int
+    num_shards: int = 1
+    shard_id: int = 0
+    device: str = 'cpu'
+    cache_to_arrow: bool = False
+    output_file_name: str = "results"
+
+
+@dataclass
+class TextPipelineConfig(PipelineConfig):
+    """
+    Configuration class for text pipelines.
+
+    Attributes:
+        encoder_model (str): The name or path of the model to be used for encoding texts into embeddings.
+        decoder_model (str): The name or path of the model to be used for decoding embeddings back into texts.
+        source_lang (str): The source language code for the texts to be encoded.
+        target_lang (str): The target language code for the texts to be decoded.
+        columns (List[str]): The columns of the dataset to process. Defaults to ["text"].
+        seeds (List[int]): Random seeds for generating different translation samples.
+    """
+    encoder_model: str
+    decoder_model: str
```

Comment:
in many cases we may want to do only encoding without decoding (which is very costly)
so I would suggest to make it optional:
```suggestion
    decoder_model: Optional[str]
```